agent.js

```diff
@@ -59,15 +59,37 @@ function checkNoProxy (uri) {
   return false
 }
 
+function getProcessEnv (env) {
+  if (!env) { return }
+
+  const processEnv = Object.assign({}, process.env) // process.env calls are slow
```

Comment:
I'm not sure this would save that much all the time: forcing the _entire_ env to get serialized on every `getProcessEnv` and all. I think it's best to just access `process.env` directly, tbh.